player/audio.c

```diff
@@ -442,6 +442,9 @@ static int reinit_audio_filters_and_output(struct MPContext *mpctx)
                           opts->audio_output_channels.num_chmaps);
     }
 
+    if (!has_video_track(mpctx))
+        ao_flags |= AO_INIT_MEDIA_ROLE_MUSIC;
```

Comment:
duplicating this seems kinda unfortunate, isn't there a better way?

Reply:
Yes it is. I wasn't able to come up with one.
It needs to be available during ao->init but that has no access to the core

Reply:
Then couldn't this be considered a bug/shortcoming in wireplumber?

Reply:
I don't think so. To be used for channel save-restore the properties have to be available as soon as possible.
Wireplumber can't mess with playback after it has started.